here-naksha-lib-view/src/main/java/com/here/naksha/lib/view/concurrent/ParallelQueryExecutor.java

```diff
@@ -116,9 +122,20 @@ private <FEATURE, CODEC extends FeatureCodec<FEATURE, CODEC>> Stream<ViewLayerRo
 
     try (MutableCursor<FEATURE, CODEC> cursor =
         session.execute(clonedRequest).mutableCursor(codecFactory)) {
+      featureCnt = cursor.asList().size();
```

Comment:
we should try to avoid calling asList() multiple times, as it is implementation driven. Better approach can be we call it once and assign it to a List variable and use the variable multiple times.

Reply:
Fixed It.